feat: New Profile Modal

caybro opened the pull request:
Fixes: #7360
Fixes: #7633
Fixes: #7597
Fixes: #7471
Fixes: #7595
Fixes: #7307
Fixes: #7634
Fixes: #7476
Fixes: #6362

### What does the PR do

Implements new Profile Modal according to design specs at https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?node-id=733%3A12552

Extracts common helper popups (rename, block, unblock, contact request, contact verification) to Global/AppMain so that they're reusable from everywhere, w/o opening the ProfileDialog

### Affected areas

ProfileDialog, AppMain, Global

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Own profile:
![Snímek obrazovky z 2022-10-10 11-02-37](https://user-images.githubusercontent.com/5377645/194831854-5689f034-69cf-4dee-b682-0cc3815183f4.png)

Contact (ENS) profile, context menu:
![Snímek obrazovky z 2022-10-10 11-02-49](https://user-images.githubusercontent.com/5377645/194831950-dd2fcf4f-01d4-46ce-a618-c537e3aa92fe.png)

Contact with nickname, dark mode:
![Snímek obrazovky z 2022-10-10 11-03-16](https://user-images.githubusercontent.com/5377645/194831900-6c068600-8b14-42d1-be42-69b0b85230e2.png)

Own profile preview: 
![Snímek obrazovky z 2022-10-10 11-05-48](https://user-images.githubusercontent.com/5377645/194832212-1f64618f-a2e4-4ba8-a40d-30fa0fdaeaa0.png)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e4980db8 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux-cpp/job/PR-7621/1/) | 2022-09-27 21:35:07 | ~4 min | `linux-cpp` | [:package:`pkg`](https://ci.status.im/job/status-desktop/job/branches/job/linux-cpp/job/PR-7621/1//consoleText) |

alexandraB99 commented:
@caybro isn't this fixing also https://github.com/status-im/status-desktop/issues/7476?

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -84,13 +84,19 @@ Item {
                 })
             return downloadPage
         }
+
```
Yup, I noticed you were doing the same; however, not everything follows this pattern so feel free to create an issue to refactor this later :)

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -164,17 +170,17 @@ Item {
+            id: profilePopup
+            profileStore: appMain.rootStore.profileSectionStore.profileStore
+            contactsStore: appMain.rootStore.profileSectionStore.contactsStore
+            onClosed: {
```
It destroys itself and this is the only place from which it is created

caybro replied:
Fixed

caybro commented on ui/imports/utils/Global.qml:
```diff
@@ -60,8 +69,56 @@ Item {
+        }
+    }
+
+    function openProfilePopup(publicKey, parentPopup) {
```
Yeah, I agree, it could be refactored (same reasoning as above); I decided to keep for backwards compatibility

caybro commented on ui/imports/shared/controls/chat/menuItems/MuteChatMenuItem.qml:
```diff
@@ -5,6 +5,6 @@ import StatusQ.Popups 0.1
     property bool muted: false
 
-    text: !muted ? qsTr("Mute chat") : qsTr("Unmute chat")
+    text: !muted ? qsTr("Mute Chat") : qsTr("Unmute Chat")
```
Yeah, will fix

caybro commented on ui/imports/shared/popups/NicknamePopup.qml:
```diff
@@ -18,15 +18,18 @@ StatusModal {
     header.title: qsTr("Nickname")
+    header.subTitleElide: Text.ElideMiddle
+
+    /*required*/ property string publicKey
```
It is used in AppMain.qml:

```qml
onOpenNicknamePopupRequested: {
            Global.openPopup(nicknamePopupComponent, {publicKey: publicKey, nickname: nickname, "header.subTitle": subtitle})
        }
```

and then later:
```qml
onEditDone: {
    if (nickname !== newNickname) {
        appMain.rootStore.contactStore.changeContactNickname(publicKey, newNickname)
        Global.nickNameChanged(publicKey, newNickname)
    }
    close()
}
```

caybro commented on ui/imports/shared/popups/OutgoingContactVerificationRequestPopup.qml:
```diff
@@ -0,0 +1,114 @@
+    title: qsTr("Verify %1's Identity").arg(root.verificationResponseDisplayName)
+
+    onClosed: {
+        root.destroy()
```
It's fine, we only instantiate it from Global.qml

caybro replied:
Ok, will fix

caybro commented on ui/imports/utils/Global.qml:
```diff
@@ -60,8 +69,56 @@ Item {
+
+    function openOutgoingIDRequestPopup(publicKey) {
+        try {
+            let verificationDetails = appMain.rootStore.profileSectionStore.contactsStore.getSentVerificationDetailsAsJson(publicKey)
```
Ok

caybro replied:
Fixed

caybro commented on ui/imports/shared/popups/ProfileDialog.qml:
```diff
@@ -0,0 +1,34 @@
+    footer: null
+
+    onClosed: {
+        root.destroy()
```
Same answer, only created from Global :)

caybro replied:
Fixed

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,664 @@
+            return Constants.userLinkPrefix + user
+        }
+
+        readonly property var conns: Connections {
```
QtObject can only contain properties, not children objects; so yes, this is a way to make the Connections private :)

caybro replied:
Right, in that case you're correct; it's just a habit to keep all internal logic inside the private `d` QtObject

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,664 @@
+            enabled: !root.readOnly
+            onClicked: {
+                Global.changeAppSectionBySectionType(Constants.appSection.profile)
+                root.close()
```
Ah right, will fix! (This happened during the extraction of the ProfileDialogView into a separate component, previously it was the Dialog)

caybro replied:
Fixed

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,664 @@
+                    visible: text
+                }
+                EmojiHash {
+                    Layout.topMargin: 4
```
Yup, that's just a visual thing, it needs some extra space besides the column spacing

caybro replied:
Ah right, will fix! Thanks for spotting

caybro replied:
Fixed

caybro commented on ui/imports/utils/Global.qml:
```diff
@@ -60,8 +69,56 @@ Item {
+
+    function openIncomingIDRequestPopup(publicKey) {
+        try {
+            let request = appMain.rootStore.profileSectionStore.contactsStore.getVerificationDetailsFromAsJson(publicKey)
```
Fixed

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,664 @@
+                Layout.leftMargin: 4
+                Layout.alignment: Qt.AlignTop
+                spacing: 4
+                Item {
```
Not sure either, none of them are what I'd call elegant :)

osmaczko reviewed (commented):
Looks good, lot of changes and simplifications :+1:

caybro commented on ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml:
```diff
@@ -10,6 +10,8 @@ import StatusQ.Core.Theme 0.1
 Dialog {
     id: root
 
+    property string subtitle
```
Yeah it seemed like a very much needed API, just because setting the subtitle is very commonplace. Plus you don't have to override the whole `background` (including hooking up the close button) just to display the subtitle. It's orthogonal to the `title` property

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -954,6 +961,44 @@ Item {
+                onEditDone: {
+                    if (nickname !== newNickname) {
+                        appMain.rootStore.contactStore.changeContactNickname(publicKey, newNickname)
+                        Global.nickNameChanged(publicKey, newNickname)
```
I agree... this is existing code though, just moved around, and it's called from this place only

caybro commented on ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:
```diff
@@ -16,7 +16,7 @@ QtObject {
     property var profileSectionModuleInst: profileSectionModule
 
-    property bool fetchingUpdate: aboutModule.fetching
+    property bool fetchingUpdate: aboutModuleInst.fetching
```
Done

caybro commented on ui/imports/shared/popups/OutgoingContactVerificationRequestPopup.qml:
```diff
@@ -0,0 +1,110 @@
+
+    signal cancelVerificationRequest(string userPublicKey)
+    signal verifiedUntrustworthy(string userPublicKey)
+    signal verifiedTrusted(string userPublicKey)
```
Done

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,665 @@
+    property var profileStore
+    property var contactsStore
+
+    signal close()
```
Done, changed the signal to `closeRequested` though

micieslak reviewed (changes requested):
Huge work :) I'm leaving several comments, and switching to testing in practice :)

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -164,17 +170,18 @@ Item {
-        onClosed: {
-            if  (profilePopup.parentPopup) {
-                profilePopup.parentPopup.close();
+    property Component profilePopupComponent: Component {
```
I don't know.. but yeah, I'll just define them as free standing Components

caybro replied:
Done

caybro commented on ui/imports/shared/panels/ModuleWarning.qml:
```diff
@@ -134,7 +131,6 @@ Item {
                 text: root.text
                 font.pixelSize: 13
                 font.weight: Font.Medium
-                anchors.verticalCenter: parent.verticalCenter
```
> The default value is 0, which means it will be Qt.AlignVCenter | Qt.AlignLeft.

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,665 @@
+
+        readonly property var conns: Connections {
+            target: Global
+            ignoreUnknownSignals: true
```
Right... it shouldn't be needed in this case

caybro replied:
Done

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,665 @@
+Pane {
+    id: root
+
+    property var parentPopup
```
Already removed by request of Patryk :)

caybro commented on ui/imports/shared/popups/ProfileDialog.qml:
```diff
@@ -0,0 +1,31 @@
+StatusDialog {
+    id: root
+
+    property var parentPopup
```
Hmm....

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -164,17 +170,18 @@ Item {
-            if  (profilePopup.parentPopup) {
-                profilePopup.parentPopup.close();
+    property Component profilePopupComponent: Component {
+        ProfileDialog {
```
Yeah, could be done this way, yup

caybro replied:
Actually not, it's called from many places via:
```qml
Global.openProfilePopup(publicKey, parentPopup)
```

caybro commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,665 @@
+                    font.pixelSize: 12
+                    color: Theme.palette.baseColor1
+                    text: {
+                        var result = ""
```
Done

micieslak reviewed (changes requested):
I've found some inconsistencies in comparison to the design and other minor issues:

I don't know if it's intentional but there is no separator line between header and content in the design:
![Screenshot from 2022-10-11 14-58-56](https://user-images.githubusercontent.com/20650004/195099217-e15c8478-29ec-4277-904e-7d81bfdb31a0.png)

There is extra space between blue handler and gray page:
![Screenshot from 2022-10-11 14-58-38](https://user-images.githubusercontent.com/20650004/195099386-8cd49a22-633a-4ec5-9f5b-603fb2f599dc.png)

According to the design qr code should have additional frame:
![Screenshot from 2022-10-11 14-56-17](https://user-images.githubusercontent.com/20650004/195099524-22ba9870-60a5-4994-a6ab-e1d2893a14a6.png)

For some window heights, textinput border is slightly broken (upper line is thicker):
![image](https://user-images.githubusercontent.com/20650004/195101454-e85816dc-f0d6-410c-b351-1911413325b5.png)

Profile preview: copy for emojis is missing:
General: size of "Link To Profile" and "Emoji Hash" is different than in design:
![Screenshot from 2022-10-11 15-19-23](https://user-images.githubusercontent.com/20650004/195102197-4e7ed2ae-3609-40cd-8551-56870d26a7de.png)

Profile preview doesn't work as expected:

[Kazam_screencast_00046.webm](https://user-images.githubusercontent.com/20650004/195104088-efe20df8-4db4-451a-8482-4b979f49330c.webm)

Opening links doesn't work, links should be centered:
![Screenshot from 2022-10-11 15-29-14](https://user-images.githubusercontent.com/20650004/195104282-825ea6e9-1d3e-4cac-b540-d33eeaabd9b2.png)

caybro commented:
> I've found some inconsistencies in comparison to the design and other minor issues:
> 
> <snip>

@micieslak thanks for the review, those are valid points, gonna fix them! 

> Opening links doesn't work, links should be centered: ![Screenshot from 2022-10-11 15-29-14](https://user-images.githubusercontent.com/20650004/195104282-825ea6e9-1d3e-4cac-b540-d33eeaabd9b2.png)

@micieslak existing code (and issue), can you pls file a bug against `SocialLinkPreview` in `ProfileBioSocialsPanel`? Thx

micieslak commented:
> > I've found some inconsistencies in comparison to the design and other minor issues:
> > I don't know if it's intentional but there is no separator line between header and content in the design: ![Screenshot from 2022-10-11 14-58-56](https://user-images.githubusercontent.com/20650004/195099217-e15c8478-29ec-4277-904e-7d81bfdb31a0.png)
> 
> @micieslak the divider is supposed to be there (one of the late changes): ![image](https://user-images.githubusercontent.com/5377645/195125679-7e241571-a753-437a-ad0f-e6d9ab0ee7d8.png)

Hmm, for me, looking at the design, the divider is visible only when scrolling. @benjthayer could you please clarify that?

noeliaSD commented on ui/imports/shared/views/ProfileDialogView.qml:
```diff
@@ -0,0 +1,663 @@
+                highlighted: moreMenu.opened
+                onClicked: moreMenu.popup(-moreMenu.width + width, height + 4)
+
+                StatusPopupMenu {
```
Just a suggestion: It could be written in a separate file.

caybro replied:
Yup, fully agree; right now it's a bit of a mess, it should be extracted and also used instead of `MessageContextMenuView` when clicking on a profile in the chat (and other parts of the app). Please file an issue :)

noeliaSD replied:
Here the new issue to keep tracking: https://github.com/status-im/status-desktop/issues/7886

noeliaSD reviewed (commented):
- Usability question: How can I get my chat key on this new design?

- I left some `UI-PixelPerfect` minor comments here:

**OWN PROFILE:**
1. Should profile link text color be black instead of gray?
2. Should profile link text have a bigger right margin respect `Copy` button?
3. QR seems not to be aligned the same way as in the design (apart of missing border mentioned before).
4. Are `Link to Profile` and `Emoji Hash` texts of size 13? They seem bigger than in the design.

**CONTACT PROFILE:**
1. Contact icon (trust icon) is smaller than in the design.
2. Should media info be horizontally aligned? I can see them a little bit moved to the left.

![Screenshot 2022-10-11 at 17 34 26](https://user-images.githubusercontent.com/97019400/195139601-efda7073-706b-423c-ac3f-b664c18c8b71.png)

**PREVIEW PROFILE:**
1. Should the `Edit Profile` button be present and disabled here? From my point of view, if I'm not missing something, it doesn't make so much sense to show it here as you are already in `Edit Profile`

caybro commented:
Thanks for the feedback @noeliaSD, comments below:

> * Usability question: How can I get my chat key on this new design?

The context (three dots) menu, "Link to profile". Yes, chatkey is not available per se, design decision

> * I left some `UI-PixelPerfect` minor comments here:
> 
> **OWN PROFILE:**
> 
> 1. Should profile link text color be black instead of gray?
> 2. Should profile link text have a bigger right margin respect `Copy` button?

Yes, and yes, will try to fix (kinda hard with StatusInput :/

> 3. QR seems not to be aligned the same way as in the design (apart of missing border mentioned before).

I just pushed a visual update to the QR code, pls check again

> 4. Are `Link to Profile` and `Emoji Hash` texts of size 13? They seem bigger than in the design.

Nope, they are 15 and can't be changed with StatusInput :/

> 
> **CONTACT PROFILE:**
> 
> 1. Contact icon (trust icon) is smaller than in the design.

Ah right, will fix (this has the tiny icons, coming for the chat)

> **PREVIEW PROFILE:**
> 
> 1. Should the `Edit Profile` button be present and disabled here? From my point of view, if I'm not missing something, it doesn't make so much sense to show it here as you are already in `Edit Profile`

Yes, should be there and disabled. In the future, the preview will gain the "aspect" combobox

elina2015 commented:
@caybro thank you for the great work done! There are some small issues:
1. Profile: 
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/14942081/195249085-7f4f706f-c8d3-41d9-84f9-44e24a8c89f4.png">

elina2015 reviewed (changes requested):
Some issues were found

caybro commented:
> Thanks for the feedback @noeliaSD, comments below:
> 
> > * Usability question: How can I get my chat key on this new design?
> 
> The context (three dots) menu, "Link to profile". Yes, chatkey is not available per se, design decision
> 
> > * I left some `UI-PixelPerfect` minor comments here:
> > 
> > **OWN PROFILE:**
> > 
> > 1. Should profile link text color be black instead of gray?
> > 2. Should profile link text have a bigger right margin respect `Copy` button?
> 
> Yes, and yes, will try to fix (kinda hard with StatusInput :/
> 
> > 3. QR seems not to be aligned the same way as in the design (apart of missing border mentioned before).
> 
> I just pushed a visual update to the QR code, pls check again
> 
> > 4. Are `Link to Profile` and `Emoji Hash` texts of size 13? They seem bigger than in the design.
> 
> Nope, they are 15 and can't be changed with StatusInput :/
> 
> > **CONTACT PROFILE:**
> > 
> > 1. Contact icon (trust icon) is smaller than in the design.
> 
> Ah right, will fix (this has the tiny icons, coming for the chat)
> 
> > **PREVIEW PROFILE:**
> > 
> > 1. Should the `Edit Profile` button be present and disabled here? From my point of view, if I'm not missing something, it doesn't make so much sense to show it here as you are already in `Edit Profile`
> 
> Yes, should be there and disabled. In the future, the preview will gain the "aspect" combobox

@noeliaSD all the issues should be addressed, please take another look, thx

noeliaSD commented:
> > Thanks for the feedback @noeliaSD, comments below:
> > > * Usability question: How can I get my chat key on this new design?
> > 
> > 
> > The context (three dots) menu, "Link to profile". Yes, chatkey is not available per se, design decision
> > > * I left some `UI-PixelPerfect` minor comments here:
> > > 
> > > **OWN PROFILE:**
> > > 
> > > 1. Should profile link text color be black instead of gray?
> > > 2. Should profile link text have a bigger right margin respect `Copy` button?
> > 
> > 
> > Yes, and yes, will try to fix (kinda hard with StatusInput :/
> > > 3. QR seems not to be aligned the same way as in the design (apart of missing border mentioned before).
> > 
> > 
> > I just pushed a visual update to the QR code, pls check again
> > > 4. Are `Link to Profile` and `Emoji Hash` texts of size 13? They seem bigger than in the design.
> > 
> > 
> > Nope, they are 15 and can't be changed with StatusInput :/
> > > **CONTACT PROFILE:**
> > > 
> > > 1. Contact icon (trust icon) is smaller than in the design.
> > 
> > 
> > Ah right, will fix (this has the tiny icons, coming for the chat)
> > > **PREVIEW PROFILE:**
> > > 
> > > 1. Should the `Edit Profile` button be present and disabled here? From my point of view, if I'm not missing something, it doesn't make so much sense to show it here as you are already in `Edit Profile`
> > 
> > 
> > Yes, should be there and disabled. In the future, the preview will gain the "aspect" combobox
> 
> @noeliaSD all the issues should be addressed, please take another look, thx

Thanks for the fixes! They look good!

Just a small pending thing:

**CONTACT PROFILE:**
- Should media info be horizontally aligned? I can see them a little bit moved to the left.

caybro commented:
> * Should media info be horizontally aligned? I can see them a little bit moved to the left.

@noeliaSD yes, [StatusCenteredFlow](https://github.com/status-im/status-desktop/commit/174f33cda3b7162bd35fdfa97848906a9571aa8f) ftw :) Gonna integrate it later today

caybro commented:
> > * Should media info be horizontally aligned? I can see them a little bit moved to the left.
> 
> @noeliaSD yes, [StatusCenteredFlow](https://github.com/status-im/status-desktop/commit/174f33cda3b7162bd35fdfa97848906a9571aa8f) ftw :) Gonna integrate it later today

@noeliaSD done, pls check again :)

caybro commented:
> Some issues were found

@elina2015 pls give it another try, thanks :)

elina2015 commented:
@caybro thank you for the fixes! There are just small issues found:
<img width="1348" alt="image" src="https://user-images.githubusercontent.com/14942081/195733835-39503b6c-0b97-4944-b2f8-675dfd021744.png">
1. The space between the Bio text and the left edge of the modal
2. The text of the Bio should be bigger and the space between lines should be a little bigger
3.

elina2015 reviewed (changes requested):
Need small fixes

caybro commented:
Thanks for the final touches @elina2015 :)

> @caybro thank you for the fixes! There are just small issues found: <img alt="image" width="1348" src="https://user-images.githubusercontent.com/14942081/195733835-39503b6c-0b97-4944-b2f8-675dfd021744.png">
> 
> 1. The space between the Bio text and the left edge of the modal

Fixed

> 2. The text of the Bio should be bigger and the space between lines should be a little bigger

Fixed

> 3. 'Link to Profile' should be 'Link to profile' (I mean, without capital 'P' for profile)
>    Could you kindly check?

Not fixed; it should be title case, just like "Emoji Hash" below

noeliaSD commented:
> 

Reviewed the social tags alignment! Thanks! Good job!

micieslak reviewed (changes requested):
Looks that we are almost there, but I still found some problems:

1. Unnecessary white space in bottom part:
![Screenshot from 2022-10-14 13-23-35](https://user-images.githubusercontent.com/20650004/195837491-fb045955-3a72-4335-98b3-556e2cf5dfe0.png)
2. Minor: social links are too short to display event short links reasonably:
![Screenshot from 2022-10-14 13-17-56](https://user-images.githubusercontent.com/20650004/195837571-1a2adb8a-6a12-4847-ae05-c364bbb92849.png)

3. Space under description in preview tends to be incorrect:
![Screenshot from 2022-10-14 13-40-56](https://user-images.githubusercontent.com/20650004/195839067-9eace063-c07e-425e-8f94-3e53e690b7f5.png)

4. Updating preview seems to be broken:

[Kazam_screencast_00058.webm](https://user-images.githubusercontent.com/20650004/195839912-69bbba72-ad9f-49d6-9aa8-be34c31e69dd.webm)

5. Minor (propably for a separate issue): It's impossible to save changes in some cases:

[Kazam_screencast_00059.webm](https://user-images.githubusercontent.com/20650004/195840317-eb624b70-92f4-4c0a-af7d-4f83a37d321e.webm)

6. There is a warning printed from time to time:
```
qrc:/imports/shared/popups/ProfileDialog.qml:23:18: QML ProfileDialogView: Binding loop detected for property "implicitHeight"
```
7. The last one: strange issue causing that fonts/edges become blurry:
 
Blurry - when window is maximized, dialog has more space than required:
![Screenshot from 2022-10-14 13-54-41](https://user-images.githubusercontent.com/20650004/195841155-097d09ed-bde4-40aa-b179-b5a47cb52abe.png)
Not blurry - when window height is limited and dialog is squeezed: 
![Screenshot from 2022-10-14 13-54-50](https://user-images.githubusercontent.com/20650004/195841158-bff8f8d1-b382-4f76-ae0c-6418ee8300ec.png)

When printing popup `y` property it shows:
```
qml: y: 64 <-- here dialog is squeezed, y pos is fixed and is integral, looks ok
qml: y: 65.5 <-- y pos is fractional, looks blury
qml: y: 66.5
qml: y: 68.5
qml: y: 70.5
qml: y: 72.5
```
It's seems to be general problem, I observed it for another dialogs as well. It may be even Qt bug, because `y` pos relies on that in `StatusDialog.qml`:
```
    anchors.centerIn: Overlay.overlay
    margins: 64
```
In `Item` it's resolved by property `anchors.alignWhenCentered` which is false by default preventing fractional positioning. It's not present in `Dialog`. Generally, probably subject of a separate issue.

caybro commented:
> Looks that we are almost there, but I still found some problems:
> 
> 1. Unnecessary white space in bottom part:
>    ![Screenshot from 2022-10-14 13-23-35](https://user-images.githubusercontent.com/20650004/195837491-fb045955-3a72-4335-98b3-556e2cf5dfe0.png)

can't reproduce

> 2. Minor: social links are too short to display event short links reasonably:
>    ![Screenshot from 2022-10-14 13-17-56](https://user-images.githubusercontent.com/20650004/195837571-1a2adb8a-6a12-4847-ae05-c364bbb92849.png)

Indeed, please file an issue for `SocialLinkPreview` component; the way this component works needs to be fixed

> 3. Space under description in preview tends to be incorrect:
>    ![Screenshot from 2022-10-14 13-40-56](https://user-images.githubusercontent.com/20650004/195839067-9eace063-c07e-425e-8f94-3e53e690b7f5.png)

can't reproduce

> 4. Updating preview seems to be broken:
> 
>  Kazam_screencast_00058.webm 

Ah, you pressed the "Save" button while already at the Preview page, will try to fix

> 5. Minor (propably for a separate issue): It's impossible to save changes in some cases:
> 
>  Kazam_screencast_00059.webm 
> 6. There is a warning printed from time to time:
> 
> ```
> qrc:/imports/shared/popups/ProfileDialog.qml:23:18: QML ProfileDialogView: Binding loop detected for property "implicitHeight"

can't reproduce (looks like a Qt issue)

> ```
> 
> 7. The last one: strange issue causing that fonts/edges become blurry:
> 
> Blurry - when window is maximized, dialog has more space than required: ![Screenshot from 2022-10-14 13-54-41](https://user-images.githubusercontent.com/20650004/195841155-097d09ed-bde4-40aa-b179-b5a47cb52abe.png) Not blurry - when window height is limited and dialog is squeezed: ![Screenshot from 2022-10-14 13-54-50](https://user-images.githubusercontent.com/20650004/195841158-bff8f8d1-b382-4f76-ae0c-6418ee8300ec.png)
> 
> When printing popup `y` property it shows:
> 
> ```
> qml: y: 64 <-- here dialog is squeezed, y pos is fixed and is integral, looks ok
> qml: y: 65.5 <-- y pos is fractional, looks blury
> qml: y: 66.5
> qml: y: 68.5
> qml: y: 70.5
> qml: y: 72.5
> ```
> 
> It's seems to be general problem, I observed it for another dialogs as well. It may be even Qt bug, because `y` pos relies on that in `StatusDialog.qml`:

again, can't reproduce, looks like a Qt issue?

caybro commented:
> > 4. Updating preview seems to be broken:
> > 
> > Kazam_screencast_00058.webm
> 
> Ah, you pressed the "Save" button while already at the Preview page, will try to fix

@micieslak fixed

elina2015 reviewed (approved):
Tested and approved

micieslak reviewed (approved):
Good job! :rocket:
